.github/workflows/build.yaml

```diff
@@ -11,6 +11,31 @@ on:
 
 jobs:
 
+  benchmark:
+     name: Run Benchmark
+     runs-on: ubuntu-latest
+     steps:
+       - uses: actions/checkout@v2
+       - name: Setup go
+         uses: actions/setup-go@v1
+         with:
+           go-version: '1.18.1' 
+       - name: Run benchmark
+         run: go test -bench . -benchmem ./... | tee output.txt
+
+       - name: Store benchmark result
+         uses: benchmark-action/github-action-benchmark@v1
```

Comment:
The fact an action exists for this makes me feel a bit better, but I'd still like to see what performance guarantees github provides in actions.

@james-milligan did you happen to find any info on this?

Reply:
I think this is the best we can do without a paid service. It at least provides a decent baseline for us to use.